src/control/server/mgmt_system.go

```diff
@@ -707,65 +709,110 @@ func (svc *mgmtSvc) SystemStart(ctx context.Context, pbReq *mgmtpb.SystemStartRe
 	return pbResp, nil
 }
 
-// SystemResetFormat implements the method defined for the Management Service.
-//
-// Prepare to reformat DAOS system by resetting format state of each rank
-// and await storage format on each relevant instance (system member).
-//
-// This control service method is triggered from the control API method of the
-// same name in lib/control/system.go and returns results from all selected ranks.
-func (svc *mgmtSvc) SystemResetFormat(ctx context.Context, pbReq *mgmtpb.SystemResetFormatReq) (*mgmtpb.SystemResetFormatResp, error) {
-	if err := svc.checkLeaderRequest(pbReq); err != nil {
+// ClusterEvent management service gRPC handler receives ClusterEvent requests
+// from control-plane instances attempting to notify the MS of a cluster event
+// in the DAOS system (this handler should only get called on the MS leader).
+func (svc *mgmtSvc) ClusterEvent(ctx context.Context, req *sharedpb.ClusterEventReq) (*sharedpb.ClusterEventResp, error) {
+	if err := svc.checkLeaderRequest(req); err != nil {
 		return nil, err
 	}
-	svc.log.Debug("Received SystemResetFormat RPC")
 
-	// We can't rely on the db being up and running, as one of the
-	// use cases for this command is to nuke the system from orbit
-	// regardless of what state it's in. But we should at least enforce
-	// that the RPC is being handled on a MS replica.
-	if err := svc.sysdb.CheckReplica(); err != nil {
+	// indicate to handler that event has been forwarded
+	resp, err := svc.events.HandleClusterEvent(req, true)
+	if err != nil {
+		return nil, errors.Wrapf(err, "handle cluster event %+v", req)
+	}
+
+	return resp, nil
+}
+
+// eraseAndRestart is called on MS replicas to shut down the raft DB and
+// remove its files before restarting the control plane server.
+func (svc *mgmtSvc) eraseAndRestart(pause bool) error {
+	svc.log.Infof("%s pid %d: erasing system db", build.ControlPlaneName, os.Getpid())
+
+	if err := svc.sysdb.Stop(); err != nil {
+		return errors.Wrap(err, "failed to stop system database")
+	}
+	if err := svc.sysdb.RemoveFiles(); err != nil {
+		return errors.Wrap(err, "failed to remove system database")
+	}
+
+	myPath, err := os.Readlink("/proc/self/exe")
+	if err != nil {
+		return errors.Wrap(err, "unable to determine path to self")
+	}
+
+	go func() {
+		if pause {
+			time.Sleep(50 * time.Millisecond)
+		}
+		if err := unix.Exec(myPath, append([]string{myPath}, os.Args[1:]...), os.Environ()); err != nil {
```

Comment:
eek! I suppose there's no real alternative? don't we need to exit the current process after spawning a new one?

Reply:
Not with the current codebase. Changes to accommodate cleanly stopping/wiping/restarting the system db are too invasive for now. After your server restructuring lands it will be easier to do the correctly. There's no need to exit, as Exec() replaces the current process.